src/xdsl/translation.py

```diff
@@ -0,0 +1,126 @@
+from xdsl.dialects.irdl import *
+from xdsl.dialects.builtin import *
+from xdsl.xdsl_opt_main import *
+from xdsl.printer import Printer
+# from xdsl.mlir_converter import MLIRConverter
+
+
+class Translation:
+
+    ctx: MLContext
+
+    def __init__(self):
+        self.ctx = MLContext()
+        self.register_the_dialects()
+        """
+        To test AnyTypeConstraint and DynTypeBaseConstraintAttr,
+        replace file to be opened with 'tests/filecheck/translation_ops.xdsl'
+        """
+        file = open("tests/filecheck/cmath.xdsl")
+
+        my_op = self.parse_xdsl_file(file)
+        my_op.walk(lambda di: self.print_di_definition(di)
+                   if isinstance(di, DialectOp) else None)
+
+    def parse_xdsl_file(self, f: IOBase):
+        input_str = f.read()
+        parser = Parser(self.ctx, input_str)
+        module = parser.parse_op()
+        if not (isinstance(module, ModuleOp)):
+            raise Exception("Expected module or program as toplevel operation")
+        return module
+
+    def register_the_dialects(self):
+        irdl = IRDL(self.ctx)
+        builtin = Builtin(self.ctx)
+
+    def print_type_definition(self, type: TypeOp):
+        print(f"  {type.name} {type.attributes['type_name'].data} {{")
+        type.walk(lambda param: self.print_parameters_definition(param)
+                  if isinstance(param, ParametersOp) else None)
+        print(f"  }}")
+
+    def print_attr_constraint(self, f):
+        if isinstance(f, AnyOfTypeConstraintAttr):
+            print(f"AnyOf", end="<")
+            for i in range(len(f.params.data) - 1):
+                self.print_attr_constraint(f.params.data[i])
+                print(", ", end="")
+            self.print_attr_constraint(f.params.data[len(f.params.data) - 1])
+            print(">", end="")
+        elif isinstance(f, EqTypeConstraintAttr):
+            p = Printer()
+            # converter = MLIRConverter(self.ctx)
+            # print(converter.convert_type(f.type))
+            p.print(f.type)
+        elif isinstance(f, AnyTypeConstraintAttr):
+            print(f"Any", end="")
+        elif isinstance(f, DynTypeBaseConstraintAttr):
+            print(f.type_name.data, end='')
+        elif isinstance(f, DynTypeParamsConstraintAttr):
+            print(f.type_name.data, end='')
+            print("<", end="")
+            for i in range(len(f.params.data)):
+                self.print_attr_constraint(f.params.data[i])
+            print(">", end="")
+        elif isinstance(f, TypeParamsConstraintAttr):
+            print(f"{f.type_name}")
+        elif isinstance(f, NamedTypeConstraintAttr):
+            print(f.type_name.data, end='')
+            print(":", end=" ")
+            self.print_attr_constraint(f.params_constraints)
+        else:
+            raise Exception(f"Invalid Constraint: {type(f)} ")
+
+    def print_parameters_definition(self, param_op: ParametersOp):
+        print(f"    {param_op.name}", end="(")
```

Comment:
You can write `ParametersOp.name` here instead, this should make the code more clear.